src/cachetools/func.py

```diff
@@ -6,100 +6,223 @@
 import math
 import random
 import time
-
-try:
-    from threading import Condition
-except ImportError:  # pragma: no cover
-    from dummy_threading import Condition
+from threading import Condition, Lock
 
 from . import FIFOCache, LFUCache, LRUCache, RRCache, TTLCache
 from . import cached
 from . import keys
 
 
 class _UnboundTTLCache(TTLCache):
-    def __init__(self, ttl, timer):
-        TTLCache.__init__(self, math.inf, ttl, timer)
+    def __init__(self, *, ttl, timer):
+        TTLCache.__init__(self, maxsize=math.inf, ttl=ttl, timer=timer)
 
     @property
     def maxsize(self):
         return None
 
 
-def _cache(cache, maxsize, typed):
+def _cache(*, cache, maxsize=128, typed=False, lock=None, condition=None, info=False):
+    if condition is None:
+        condition = Condition()
+
     def decorator(func):
         key = keys.typedkey if typed else keys.hashkey
-        wrapper = cached(cache=cache, key=key, condition=Condition(), info=True)(func)
+        wrapper = cached(
+            cache=cache, key=key, condition=condition, info=info, lock=lock
+        )(func)
```

Comment:
`info=True` was the default here, in contrast to the main module, which is an unnecessary slowdown for most use-cases of the decorator. changed to `False`